integrations/src/main/scala/com/quantemplate/integrations/commands/ConfigDef.scala

```diff
@@ -0,0 +1,3 @@
+package com.quantemplate.integrations.commands
+
+trait ConfigDef
```

Comment:
I would rather make it sealed trait and move all possible configurations to this file. This provides checking on compilation level in match case syntax, that all possible implementations are handled.

Reply:
In general, I agree. But here the downside to that is getting a lot of noise in the `ConfigDef.scala` from methods and serialization logic on other case classes, which should be in their respective folders.
I saw that the `circe-yaml` was upgraded to Scala 3, so maybe soon I'll be able to use the new `dervive` mechanism for encoders/decoders and this will be less of an issue.